### PR TITLE
Add support for fetching and resolving issues

### PR DESCRIPTION
**Pull request by mattvagni**
As part of this I've also moved all CI to use node 18 and pnpm.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,3 +93,8 @@ Equivalent to a 401 or 403 response. Normally means your API key doesn't exist o
 
 [(view source)](./src/error.ts)
 Equivalent to a 500 response. If this happens something unexpected within Plain happened.
+
+### UnknownError
+
+[(view source)](./src/error.ts)
+Fallback error type when something unexpected happens.
```
Comment (mattvagni) [inline README.md:100]: Realised this was missing from the readme so I documented this as well!

File: package.json [hunk 2]
```diff
@@ -37,6 +40,7 @@
     "rollup-plugin-esbuild": "^5.0.0",
     "typedoc": "^0.24.7",
     "typedoc-plugin-missing-exports": "^2.0.0",
+    "typescript": "^5.1.6",
```
Comment (mattvagni) [inline package.json:43]: Had to install typescript since pnpm is a lot stricter than npm (which is a good thing)

**Review by jesushernandez — APPROVED** [review]

File: src/client.ts [hunk 3]
```diff
@@ -168,6 +173,37 @@ export class PlainClient {
     return unwrapData(res, (q) => nonNullable(q.createIssue.issue));
   }
 
+  /**
+   * Resolve an issue for a customer.
+   */
+  async resolveIssue(
+    input: VariablesOf<typeof ResolveIssueDocument>['input']
```
Comment (jordrake) [inline src/client.ts:180]: This feels a little 'raw' for an SDK, this could just be `issueId` and we construct the input

Comment (mattvagni) [reply src/client.ts:180]: That is what it is but just as a named argument (notice the 'input' at the end):

This means this is valid:

```tsx
  const foo = await client.resolveIssue({
    issueId: 'it_123'
  })
```

Comment (mattvagni) [reply src/client.ts:180]: Or did you mean: `resolveIssue('asda')` would be nicer?

Comment (jordrake) [reply src/client.ts:180]: The latter

**Pull request merged**